### PR TITLE
[ops, dist, config, docs, ci] fix: make chunk loss compatible with activation offload

### PR DESCRIPTION
**Pull request by YLlllllllllll**
### What does this PR do?

> Make `chunk_loss` / `npu` cross-entropy work with activation offload.
>
> Previously, `ChunkLoss` used `torch.func.grad_and_value`, which rejects any
> active `saved_tensors_hooks` — including the hooks installed by
> `enable_activation`. Enabling both paths failed closed.
>
> This PR replaces that transform with an equivalent first-order
> `torch.autograd.grad` VJP, keeps the short-lived chunk CE graph on device,
> and lets only the long-lived precomputed grads go through the outer offload
> hooks. An opt-in `cross_entropy_loss_release_cache` flag can reclaim the
> transient CE allocator cache before model backward on memory-constrained
> profiles.

Replaces closed #964 after rebase onto latest `main` (fork push required recreating the head branch).

### Checklist Before Starting

- Search for relative PRs/issues and link here: supersedes #964; overlap note with #943 (`ce_loss_func` / `chunk_loss_function` nearby hunks)
- PR title follows `[{modules}] {type}: {description}` format

### Test

> - New unit tests in `tests/ops/test_chunk_loss_saved_tensor_hooks.py`
> - `python3 -m py_compile` on touched Python files
> - Validated in internal NPU training profiles that combine
>   `chunk_loss`/`npu` with `train.accelerator.offload_config.enable_activation`

### API and Usage Example

```yaml
model:
  ops_implementation:
    cross_entropy_loss_implementation: chunk_loss   # or npu
    cross_entropy_loss_release_cache: false          # set true on tight HBM profiles
train:
  accelerator:
    offload_config:
      enable_activation: true
```

### Design & Code Changes

1. **`veomni/ops/kernels/cross_entropy/chunk_loss.py`**
   - Drop `torch.func`; compute per-chunk VJP with `torch.autograd.grad`
   - Shadow outer activation-offload hooks with identity hooks for the
     short-lived CE graph so vocab-sized logits are not forced through CPU
   - After `ChunkLoss.apply`, optionally sync + `empty_cache` when
     `release_cache=true`
2. **`veomni/ops/kernels/cross_entropy/__init__.py`**
   - Thread `cross_entropy_loss_release_cache` from ops config into the
     chunk-loss dispatch
3. **`veomni/arguments/arguments_types.py` + docs**
   - Document the new opt-in flag
4. **Tests**
   - Regression coverage for hooks / offload / release_cache

This change is hardware-agnostic and does **not** include Ascend 910B4 Mojo
compat overlays, validators, or GDR pad logic.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks (targeted compile + diff --check)
- Added/updated documentation
- Added tests (`tests/ops/test_chunk_loss_saved_tensor_hooks.py`)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by YLlllllllllll** [comment]
Rebased replacement for closed #964 (fork force-push of a rebased tip is blocked by the ByteDance-author pre-receive hook, so the head branch had to be recreated).

Same patch as #964 @ `2855cc5`, now on latest `main`.

Non-blocking follow-ups from the Cursor review on #964 still apply:
- Overlap with #943 nearby hunks
- Factory `partial` vs lazy `_chunk_loss_dispatch`
- `release_cache` not covering distill/log-probs paths
- Optional design-doc note about identity hooks keeping vocab logits on-device

Happy to land those as a small follow-up after this merges.

@Coach257 please re-approve / merge when convenient — prior approval was on #964.

**Comment by YLlllllllllll** [comment]
Addressed the two non-blocking Cursor review notes:

1. **Lazy `release_cache`**: `_chunk_loss_causal_factory` now returns bare `_chunk_loss_dispatch` (no `partial` snapshot), so OpSlot and `LOSS_MAPPING` both resolve config lazily and keep a readable kernel name.
2. **`release_cache` on log-probs / distill**: dispatch now runs `_release_accelerator_cache` after those chunked helpers return; docs/help text updated accordingly. Also added the design note that identity hooks keep vocab logits on-device and `release_cache` is the mitigation.

@Coach257 ready for re-review / merge when convenient.
